### PR TITLE
enhancement:mui-tables & no-group section bug

### PR DESCRIPTION
**Pull request by kujo205**
- merged separate mobile & desktop tables into 1
- changed Alerts to Toasts
- rewrote tables using Mui Grid (#300)
- fixed a bug with not displaying `no-group` section  (#550)

### CODE REVIEW

File: src/components/pages/account-page/components/group-tab/components/table/student-table/components/EditingColumn.tsx [hunk 1]
```diff
@@ -50,16 +60,16 @@ const EditingColumn: FC<EditingColumnProps> = ({ student, refetch }) => {
...
-            : UserGroupRole.MODERATOR,
-      });
+      setChangePopupOpen(false);
+      if (user.group)
+        await GroupAPI.updateStudentRole(user?.group?.id, student.id, {
+          roleName:
+            student.role === UserGroupRole.MODERATOR
+              ? UserGroupRole.STUDENT
+              : UserGroupRole.MODERATOR,
+        });
```
Comment (ria0308) [inline src/components/pages/account-page/components/group-tab/components/table/student-table/components/EditingColumn.tsx:72]: alert

File: src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                  text={buttonName}
+                  variant={ButtonVariant.TEXT}
+                  startIcon={arrowIcon}
+                  onClick={() => setChangePopupOpen(true)}
+                />
+              )}
+            {((user.group?.role === UserGroupRole.CAPTAIN &&
+              student.role !== UserGroupRole.CAPTAIN) ||
+              (user.group?.role === UserGroupRole.MODERATOR &&
+                student.role == UserGroupRole.STUDENT)) && (
```
Comment (ria0308) [inline src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx:76]: move this logic to some const

File: src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    <ClickAwayListener onClickAway={() => setIsPopperOpen(false)}>
+      <Box>
+        <IconButton
+          ref={EllipsisIconRef}
+          icon={<EllipsisVerticalIcon className={'icon'} />}
+          color={IconButtonColor.TRANSPARENT}
+          onClick={() => setIsPopperOpen(pr => !pr)}
+          disabled={
+            user.group?.role === UserGroupRole.STUDENT ||
+            student.role === UserGroupRole.CAPTAIN
```
Comment (ria0308) [inline src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx:54]: move this logic to some const

File: src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+          }
+        ></IconButton>
+        <Popper
+          open={isPopperOpen}
+          placement={'bottom-end'}
+          anchorEl={EllipsisIconRef.current}
+        >
+          <Stack sx={styles.dropdown}>
+            {user.group?.role === UserGroupRole.CAPTAIN &&
+              student.role !== UserGroupRole.CAPTAIN && (
```
Comment (ria0308) [inline src/components/pages/account-page/components/group-tab/components/table/student-table/components/MobileDropdown.tsx:64]: move this logic to some const

File: src/components/pages/account-page/components/group-tab/components/table/student-table/StudentsTable.tsx [hunk 5]
```diff
@@ -1,66 +1,70 @@
...
+          3000,
         );
       } else {
-        dispatch(
-          showAlert({
-            title: 'Здається ти ввів неправильні значення!',
-            color: AlertColor.ERROR,
-          }),
-        );
+        toast.error('Здається ти ввів неправильні значення!', '', 3000);
```
Comment (ria0308) [inline src/components/pages/account-page/components/group-tab/components/table/student-table/StudentsTable.tsx:64]: coma after "здається"

**Review by ria0308 — CHANGES_REQUESTED** [review]

**Review by ria0308 — APPROVED** [review]

**Review by Markusplay — APPROVED** [review]

**Pull request merged**